### PR TITLE
feat(enhancement): Add support for displaying the fuel and regular ammo count properly

### PR DESCRIPTION
**Pull request by tibetiroka**
**Feature:** This PR implements the feature requested by the TODO comment.

## Feature Details
The ammo display will now show the minimum of the two ammo requirements.

## UI Screenshots
N/A

## Testing Done
Fuel-based weapons still display properly. I'm too dumb to make proper test ships/outfits.

### Automated Tests Added
N/A

## Performance Impact
N/A

### CODE REVIEW

**Review by Hurleveur — APPROVED** [review]
it's nice seeing TODOs going away

File: source/AmmoDisplay.cpp [hunk 1]
```diff
@@ -49,16 +49,17 @@ void AmmoDisplay::Update(const Ship &flagship)
 			continue;
 
 		double ammoCount = -1.;
-		// TODO: if a weapon has both an ammo requirement and consumes fuel, the ammo display will only show the ammo,
-		// not the fuel, so the weapon may not fire because it is out of fuel even though it still has ammo,
-		// and will not show a '0' in the ammo display.
 		if(secWeapon->Ammo())
 			ammoCount = flagship.OutfitCount(secWeapon->Ammo());
 		else if(secWeapon->FiringFuel())
```
Comment (quyykk) [inline source/AmmoDisplay.cpp:54]: ~~that's what happens when you don't test~~

```suggestion
		if(secWeapon->FiringFuel())
```

**Review by quyykk — COMMENTED** [review]
Can you add a comment too where you added the if/else explaining the scenario?

**Comment by quyykk** [comment]
I'm not entirely convinced we need a template for this. The primary issue for me is that GitHub doesn't provide proper UX for selecting templates, so every template we add just increases the amount of text to delete and might make it harder for people to find templates? idk

File: source/AmmoDisplay.cpp (hunk 1 above)
Comment (tibetiroka) [reply source/AmmoDisplay.cpp]: ~~oops lol~~

File: source/AmmoDisplay.cpp [hunk 2]
```diff
@@ -49,16 +49,17 @@ void AmmoDisplay::Update(const Ship &flagship)
...
+		if(secWeapon->FiringFuel())
 		{
 			double remaining = flagship.Fuel()
 				* flagship.Attributes().Get("fuel capacity");
-			ammoCount = remaining / secWeapon->FiringFuel();
+			double fuelAmmoCount = remaining / secWeapon->FiringFuel();
+			if(ammoCount == -1.)
+				ammoCount = fuelAmmoCount;
+			else
+				ammoCount = min(ammoCount, fuelAmmoCount);
```
Comment (quyykk) [inline source/AmmoDisplay.cpp:62]: Can you make this a ternary?

Comment (tibetiroka) [reply source/AmmoDisplay.cpp:64]: If you think that doesn't get too unwieldy, sure.

Comment (quyykk) [reply source/AmmoDisplay.cpp]: nvm you're right

**Review by quyykk — APPROVED** [review]

**Pull request merged**